### PR TITLE
[FAB-17266] Move CouchDB tutorial to test network

### PR DESCRIPTION
**Pull request by nikhil550**
Signed-off-by: NIKHIL E GUPTA <negupta@us.ibm.com>

#### Type of change

- Documentation update

#### Description

Move CouchDB tutorial to test network as part of moving docs away from BYFN

### CODE REVIEW

File: docs/source/couchdb_tutorial.rst [hunk 1]
```diff
@@ -910,8 +891,8 @@ use the same index name but alter the index definition. Simply edit the index
 JSON file and add or remove fields from the index. Fabric only supports the
 index type JSON, changing the index type is not supported. The updated
```
Comment (mastersingh24) [inline docs/source/couchdb_tutorial.rst]:
```suggestion
index type JSON.  Changing the index type is not supported. The updated
```

File: docs/source/couchdb_tutorial.rst [hunk 2]
```diff
@@ -930,27 +911,26 @@ line curl utility to create and update indexes.
 .. note:: The Fauxton interface is a web UI for the creation, update, and
```
Comment (mastersingh24) [inline docs/source/couchdb_tutorial.rst:911]: I know this was here before, but do we really need to have explicit doc on how to use CouchDB tools?

File: docs/source/couchdb_tutorial.rst (hunk 1 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: Will do

File: docs/source/couchdb_tutorial.rst (hunk 2 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst:911]: From past experience, Fabric users are not usually experienced database admins and may need a push in the right direction. See the best practices for queries and indexes section above, which was put in based on user feedback. It also makes the tutorial a bit more fun. You can go to your browser and see your indexes. The note is far down enough that it does not clutter the tutorial.

File: docs/source/couchdb_tutorial.rst [hunk 3]
```diff
@@ -28,28 +28,24 @@ and for more information on the Fabric ledger refer to the `Ledger <ledger/ledge
...
-key-value pairs and supports key, key range, and composite key queries only.
-CouchDB is an optional alternate state database that supports rich
-queries when chaincode data values are modeled as JSON. Rich queries are more
-flexible and efficient against large indexed data stores, when you want to
-query the actual data value content rather than the keys. CouchDB is a JSON
-document datastore rather than a pure key-value store therefore enabling
-indexing of the contents of the documents in the database.
+database embedded in the peer node. LevelDB stores chaincode data as simple
+key-value pairs and only supports key, key range, and composite key queries.
+CouchDB is an optional alternate state database that allows you to model data
```
Comment (sykesm) [inline docs/source/couchdb_tutorial.rst]:
```suggestion
CouchDB is an optional, alternate state database that allows you to model data
```

File: docs/source/couchdb_tutorial.rst [hunk 4]
```diff
@@ -28,28 +28,24 @@ and for more information on the Fabric ledger refer to the `Ledger <ledger/ledge
...
+on the ledger as JSON and issue rich queries against data values rather than
+the keys. CouchDB also allows you to deploy indexes with your chaincode to make
+queries more efficient, and enable you to query large datasets.
```
Comment (sykesm) [inline docs/source/couchdb_tutorial.rst]:
```suggestion
queries more efficient and enable you to query large datasets.
```

File: docs/source/couchdb_tutorial.rst [hunk 5]
```diff
@@ -246,24 +239,14 @@ and refreshed every time new records are added to the state database.
...
-During the chaincode installation using the ``client.installChaincode()`` API,
-include the attribute (``metadataPath``) in the `installation request <https://hyperledger.github.io/fabric-sdk-node/master/global.html#ChaincodeInstallRequest>`__.
-The value of the metadataPath is a string representing the absolute path to the
-directory structure containing the JSON index file(s).
-
-Alternatively, if you are using the
-:doc:`commands/peercommand` command to install and instantiate the chaincode, then the JSON
-index files must be located under the path ``META-INF/statedb/couchdb/indexes``
+Once you finalize an index, you need to package it with your chaincode for
+deployment by being placed alongside it in the appropriate metadata folder. You
```
Comment (sykesm) [inline docs/source/couchdb_tutorial.rst]:
```suggestion
deployment by placing it in the appropriate metadata folder. You
```

This also seems to overlap with the explicit directions that were added below. Seems we would only need one.

File: docs/source/couchdb_tutorial.rst [hunk 6]
```diff
@@ -315,39 +313,39 @@ Therefore we need to install a chaincode on every peer that will execute and
 endorse our transactions. However, before we can interact with our chaincode,
 the members of the channel need to agree on a chaincode definition that
 establishes chaincode governance. In the previous section, we demonstrated how
-to add the index to the chaincode folder so that it is ready for deployment.
+to add the index to the chaincode folder so that the index is deployed with
+the chaincode .
```
Comment (sykesm) [inline docs/source/couchdb_tutorial.rst]:
```suggestion
the chaincode.
```

File: docs/source/couchdb_tutorial.rst [hunk 7]
```diff
@@ -315,39 +313,39 @@ Therefore we need to install a chaincode on every peer that will execute and
...
+variables in your CLI to interact with the network as the Org1 admin. Make sure
+that you are in the `test-network` directory.
 
 .. code:: bash
 
-   bash-4.4#
+    export PATH=${PWD}/../bin:${PWD}:$PATH
+    export FABRIC_CFG_PATH=$PWD/../config/
+    export CORE_PEER_TLS_ENABLED=true
+    export CORE_PEER_LOCALMSPID="Org1MSP"
```
Comment (sykesm) [inline docs/source/couchdb_tutorial.rst]: nit: double quotes should be used around the `$PWD` references. I wouldn't bother changing it here but may be worth a separate task for some other time.

File: docs/source/couchdb_tutorial.rst (hunk 2 above)
Comment (mastersingh24) [reply docs/source/couchdb_tutorial.rst:911]: ok

File: docs/source/couchdb_tutorial.rst (hunk 3 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: done.

File: docs/source/couchdb_tutorial.rst (hunk 4 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: done.

File: docs/source/couchdb_tutorial.rst (hunk 5 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: done.

File: docs/source/couchdb_tutorial.rst (hunk 6 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: done.

File: docs/source/couchdb_tutorial.rst (hunk 7 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: done.

**Review by pamandrejko — COMMENTED** [review]
In this tutorial and others, should we be referring to it as the Test Network instead of the test network?

File: docs/source/couchdb_tutorial.rst [hunk 8]
```diff
@@ -282,28 +265,43 @@ Start the network
...
+From the `test-network` directory, deploy the test network with CouchDB with the
+following command:
+
+.. code:: bash
 
- This will create a simple Fabric network consisting of a single channel named
- `mychannel` with two organizations (each maintaining two peer nodes) and an
- ordering service while using CouchDB as the state database.
+    ./network.sh up createChannel -c mychannel -s couchdb
+
```
[the remark this answers is not in the document]
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst:301]: done

**Review by nikhil550 — COMMENTED** [review]
> In this tutorial and others, should we be referring to it as the Test Network instead of the test network?

I have in lower case in all of the docs, so how about we leave it lower case for now. But I can see the case for using Test Network in the future.

File: docs/source/couchdb_tutorial.rst [hunk 9]
```diff
@@ -63,16 +59,16 @@ Enable CouchDB in Hyperledger Fabric
 
 CouchDB runs as a separate database process alongside the peer, therefore there
```
Comment (mastersingh24) [inline docs/source/couchdb_tutorial.rst]:
```suggestion
CouchDB runs as a separate database process alongside the peer, there
```

File: docs/source/couchdb_tutorial.rst [hunk 10]
```diff
@@ -63,16 +59,16 @@ Enable CouchDB in Hyperledger Fabric
...
 are additional considerations in terms of setup, management, and operations.
-A docker image of `CouchDB <https://hub.docker.com/r/hyperledger/fabric-couchdb/>`__
+A Docker image of `CouchDB <https://hub.docker.com/r/hyperledger/fabric-couchdb/>`__
```
Comment (mastersingh24) [inline docs/source/couchdb_tutorial.rst]:
```suggestion
A Docker image of `CouchDB <https://hub.docker.com/_/couchdb/>`__
```
We are no longer building or shipping our own CouchDB image.

File: docs/source/couchdb_tutorial.rst [hunk 11]
```diff
@@ -81,10 +77,7 @@ FABRIC_CFG_PATH:
 
 Edit the ``stateDatabase`` section of ``core.yaml``. Specify ``CouchDB`` as the
 ``stateDatabase`` and fill in the associated ``couchDBConfig`` properties. For
-more details on configuring CouchDB to work with fabric, refer `here <http://hyperledger-fabric.readthedocs.io/en/master/couchdb_as_state_database.html#couchdb-configuration>`__.
-To view an example of a core.yaml file configured for CouchDB, examine the
-BYFN ``docker-compose-couch.yaml`` in the ``HyperLedger/fabric-samples/first-network``
-directory.
+more information, see `CouchDB configuration <http://hyperledger-fabric.readthedocs.io/en/master/couchdb_as_state_database.html#couchdb-configuration>`__.
```
Comment (mastersingh24) [inline docs/source/couchdb_tutorial.rst]: Do we need to hard-code the link to "master"?

**Review by mastersingh24 — CHANGES_REQUESTED** [review]

File: docs/source/couchdb_tutorial.rst (hunk 10 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: done.

File: docs/source/couchdb_tutorial.rst (hunk 9 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: done.

File: docs/source/couchdb_tutorial.rst (hunk 11 above)
Comment (nikhil550) [reply docs/source/couchdb_tutorial.rst]: switching to relative links.

**Review by mastersingh24 — APPROVED** [review]

**Pull request merged**